src/ping.ts

```diff
@@ -36,7 +36,7 @@ export class Ping {
       };
       this.client.logger.warn('Missed a ping.', Object.assign({ error: err }, info));
       if (++this.numberOfFailedPings > this.failedPingsBeforeDisconnect) {
-        clearInterval(this.pingIntervalId);
+        this.stop();
```

Comment:
`this.stop()` clears the interval and sets it to `null` which I thought was more inclusive.